### PR TITLE
feat: tested code examples

### PR DESCRIPTION
**Pull request by weichweich**
* load code examples from example directory
* execute all examples
* connect to peregrine and use peregrine in the workshop
* add github pipeline

### CODE REVIEW

**Comment by weichweich** [comment]
@Dudleyneedham maybe we could do the same for the other code examples?

File: docs/chain/collator.md [hunk 1]
```diff
@@ -58,12 +57,29 @@ In the following we will explain a small selection of parameters.
...
+Exposing the RPC endpoints can be done using the following parameters:
 
 ```
+--rpc-port=9933
 --rpc-cors=all
 --rpc-methods=unsafe
---rpc-external
+```
+
+With these parameters you can make RPCalls directly using curl.
```
Comment (Dudleyneedham) [inline docs/chain/collator.md:68]: `RPCalls` is a better name

File: docs/sdk/0-workshop/00-welcome.md [hunk 2]
```diff
@@ -5,13 +5,17 @@ title: 👋🏻 Welcome
...
-> 📦 Topics: <a href="https://github.com/KILTprotocol/sdk-js">KILT SDK</a> essentials, basic credential workflow without privacy enhancement. This includes creating a CTYPE and a claim; attesting a claim; and finally verifying the attested claim.
->
-> ⏳ Duration: 15-45 minutes.
->
-> 🤓 Prerequisites: basic JavaScript or TypeScript knowledge.
->
-> ❓ Questions? Join our <a href="https://discord.gg/hX4pc8rdHS">developer community channel</a>!
+:::info What you can expect
+
+📦 Topics: <a href="https://github.com/KILTprotocol/sdk-js">KILT SDK</a> essentials, basic credential workflow without privacy enhancement. This includes creating a CTYPE and a claim; attesting a claim; and finally verifying the attested claim.
```
Comment (Dudleyneedham) [inline docs/sdk/0-workshop/00-welcome.md]: Do we need the privacy enhancement in this sentence?

File: docs/sdk/0-workshop/03-identity.md [hunk 3]
```diff
@@ -10,16 +13,25 @@ In KILT, an identity is an object that represents an entity - for example a pers
...
-> - A signing keypair, used to sign claims;
-> - The address, which is generated from the signing public key;
-> - An encryption keypair, used to encrypt messages between participants of the system;
-> - A user secret, used for privacy enhanced attestations;
+
+:::info KILT Identity
+
+A KILT identity is a set of cryptographic elements:
+
+- A signing keypair, used to sign claims;
```
Comment (Dudleyneedham) [inline docs/sdk/0-workshop/03-identity.md]: Do you need to inform them that it is used to sign claims? Maybe just say the following.

used to sign transactions such as claims.

File: docs/sdk/0-workshop/03-identity.md [hunk 4]
```diff
@@ -10,16 +13,25 @@ In KILT, an identity is an object that represents an entity - for example a pers
...
+- A user secret, used for privacy enhanced attestations;
+
+:::
 
 All we need to create an identity is a mnemonic.
 
-> 💡 In cryptography, a mnemonic is a random series of words. For example, `gold upset segment cake universe` is a mnemonic. It's used to generate keypairs. What's great about a mnemonic is that it's **human-readable**. A person can memorize it, and use it later to re-generate their keypairs and address.
+:::info mnemonic
+
+In cryptography, a mnemonic is a random series of words. For example, `gold upset segment cake universe` is a mnemonic. It's used to generate keypairs. What's great about a mnemonic is that it's **human-readable**. A person can memorize it, and use it later to re-generate their keypairs and address.
```
Comment (Dudleyneedham) [inline docs/sdk/0-workshop/03-identity.md]:
```suggestion
In cryptography, a mnemonic usually consists of 12 or 24 random series of words. For example, `gold upset segment cake universe` is a mnemonic. It's used to generate keypairs. What's great about a mnemonic is that it's **human-readable**. A person can memorize it, and use it later to re-generate their keypairs and address.
```

**Review by Dudleyneedham — CHANGES_REQUESTED** [review]
Looks good overall just need some minor suggestions

**Review by Dudleyneedham — APPROVED** [review]
LGTM

**Pull request merged**